Nos/Service/EventProcessor.swift

```diff
@@ -32,6 +33,15 @@ enum EventProcessor {
                 event.isVerified = true
             }
             
+            // if this is a zap receipt, pull the zap request out of the description tag and parse it as well
+            if event.kind == EventKind.zapReceipt.rawValue,
+                let tags = event.allTags as? [[String]],
+                let descriptionTag = tags.first(where: { $0.first == "description" }),
+                let zapRequestJSONData = descriptionTag.last?.data(using: .utf8) {
+                let zapRequest = try JSONDecoder().decode(JSONEvent.self, from: zapRequestJSONData)
+                try parse(jsonEvent: zapRequest, from: relay, in: parseContext, skipVerification: skipVerification)
```

Comment:
Just to make sure I understand: we're calling parse here to make sure the zap request JSON event is stored in Core Data? I know we reference it in the notifications tab, so that would make sense.

Reply:
Exactly. Zaps are a little different because they come in two pieces, and clients don't actually receive the zap requests from relays directly. Rather, requests are sent to the lnurl callback url, and then the recipient's wallet publishes the receipt broadly to relays, with the wrapped request.